pty/src/win/mod.rs

```diff
@@ -62,11 +65,12 @@ impl Child for WinChild {
         if let Ok(Some(status)) = self.try_wait() {
             return Ok(status);
         }
+        let proc = self.proc.lock().unwrap();
         unsafe {
-            WaitForSingleObject(self.proc.as_raw_handle(), INFINITE);
+            WaitForSingleObject(proc.as_raw_handle(), INFINITE);
```

Comment:
Waiting while a mutex is held is a potential source of trouble.
There are at least two possible ways to resolve this:

* Grab the raw handle under the lock, but release it prior to the wait:

```rust
let handle = self.proc.lock().unwrap().as_raw_handle();
unsafe {
    WaitForSingleObject(handle, INFINITE);
}
```

This feels like it might be potentially possible for eg: the handle to be closed outside of the scope of the Mutex guarding it, but in practice, since `wait` requires a mutable reference it should be theoretically impossible for there to be a concurrent actor.

* Make a copy of the handle:

See https://docs.rs/filedescriptor/0.7.3/filedescriptor/struct.OwnedHandle.html#method.try_clone

```rust
let handle = self.proc.lock().unwrap().try_clone()?;
unsafe {
   WaitForSingleObject(handle.as_raw_handle(), INFINITE);
}
```

This is safer, but a little more heavy weight, and the `try_clone` could fail due to kernel resource shortage (although: the system would likely be badly borked if that were true so something else
would be likely to fail as well).